### ISSUE TITLE
Code generation issue when nested types have duplicate name

### ISSUE THREAD

**Issue by baralong** [mulesoft/amf-dotnet-tools#5]
All our responses have a similar shape:
```raml
type: object
description: ...

properties:
  Data:
    type: object
    properties: 
    ....request specific data...
  Status:
    type: object
    properties: 
      Success: boolean
      Code: string
```
However the generated code only generates one type for the `Data` object. This means we can't deserialize the responses correctly.

The source files for the generation are zipped here [rtbs-pm-sapi-1.0.21-raml.zip](https://github.com/mulesoft/amf-dotnet-tools/files/3164967/rtbs-pm-sapi-1.0.21-raml.zip)

**Comment by woodp** [comment]
@baralong thanks for the feedback, will check all of these

**Comment by woodp** [comment]
fixed in release v0.02-beta
